### DISCUSSION TITLE
Multi Agent Chat with 2 Rag Systems

### DISCUSSION THREAD

**Discussion by Jonathanpro** [microsoft/autogen#3062] [Q&A]
Hello everyone I want to create a Groupchat between two agents. Each of them has an assistant that has access to its own RAG System.

Sofar I have created a solution that somehow is related to https://microsoft.github.io/autogen/blog/2023/10/18/RetrieveChat/#integrate-with-other-agents-in-a-group-chat

When the first Agent asks its RAG system I receive a good answer however when the second agent than askes its RAG system i receive a "Error: Collection autogen-docs already exists."

I have noticed that both RAG systems use the "tmp" folder to write the embedding to.
Hence I concluded that there might be a problem that the two RAG system needs to write their embeddings into seperated folders.

For each of the RAG assistants I have written a RetrieveUserProxyAgent
that looks like this 

A_aid = RetrieveUserProxyAgent(
    name="A_aid",
    llm_config=llm_config,
    human_input_mode="NEVER",
    retrieve_config={
        "task": "qa",
        "docs_path": "path_to_file",
        # "vector_db": vectordb_A,
        # "docs_path": None,
        "vector_db": "chroma",
        "client": chromadb.PersistentClient(path="chromadb_A_aid"),
    },
    description="Assistent A",
)

for B likewise

I tried to use vector_db instead of docs_path but than the conversation seems to have no access to the knowledge of the RAG

I would welcome any suggestions !

**Comment by Josephrp** [comment]
you should try to initialize the vectordb for both agents individually , then try again with the correct parameters based on that
